### PR TITLE
Support table attribute in /update

### PR DESCRIPTION
**Pull request by graft**
This adds some basic support in the /update endpoint for updating `table` attributes. There were two main issues to resolve:

1) It is customary to use `table` to link to a leaf model with no identifier (that is, just an ancillary table attached to this model; the only relevant identifier is the parent's), in which case many Magma operations substitute the database `id` column for the identifier. The /update revision format, however, posts changes as { identifier => { identifier_changes } }. If we are inserting new records, there must be a way for us to specify a non-existent id (that is, cause the insertion to create a new id).

2) There might be existing records in a table that should be removed before inserting new values.

Here I have implemented (1) and not (2). I think it is possible to do (2) but not without increasing the coupling across several Magma classes, which I am not willing to do, and will instead opt for the somewhat more painful route of deleting records in the console manually before re-inserting (if necessary) to meet our v1 goals.

(1) is mostly accomplished by resurrecting the old `TempId` system - this allowed a way to create temporary associations by wrapping objects (e.g. the hash representing a patient) in a `TempId`, which following the insertion into the database can be updated with the real id of the insertion, allowing us to resolve links before the /update completes (see `Magma::Loader#update_temp_ids`). Here I have reattached this system to the /update input using the special identifier `::temp`, followed by any unique string, e.g. `::temp1`, `::tempkevin`, etc., which may be used to generate a `TempId`.

Regarding said coupling, the main discovery I have made here (but not implemented) is how to resolve a good deal of the confusion currently at play in the code base. I don't think I can take the time to go through all of the changes required and meet our goals, so this discovery must play out later. Summarized in #196.

### CODE REVIEW

File: spec/update_spec.rb [hunk 1]
```diff
@@ -593,237 +590,152 @@ def update(revisions, user_type=:editor)
...
-      }))
+      # the updated record is returned
+      expect(last_response.status).to eq(200)
+      #expect(json_document(:labor, 'The Golden Apples of the Hesperides')[:prize]).to match_array(Labors::Prize.select_map(:id))
     end
 
-    it 'removes an image reference using ::blank' do
-      lion = create(:monster, name: 'Nemean Lion', species: 'lion', selfie: '{"location": "https://metis.test/labors/Nemean Lion/headshot.png", "filename": "", "original_filename": ""}')
-
+    xit 'updates a table and returns the correct id' do
```
Comment (coleshaw) [inline spec/update_spec.rb:657]: Is this test supposed to be skipped?

File: spec/update_spec.rb [hunk 2]
```diff
@@ -593,237 +590,152 @@ def update(revisions, user_type=:editor)
...
-      }))
+      # the updated record is returned
+      expect(last_response.status).to eq(200)
+      expect(json_body[:models][:prize][:documents].values.map{|p|p[:name]}).to eq(['apple of joy']*2)
     end
 
-    it 'links an image from metis' do
-      Timecop.freeze(DateTime.new(500))
-      lion = create(:monster, name: 'Nemean Lion', species: 'lion')
+    xit 'replaces an existing table' do
```
Comment (coleshaw) [inline spec/update_spec.rb:708]: Is this test supposed to be skipped?

Comment (graft) [reply spec/update_spec.rb:708]: Yes, both of these should be skipped. I would like this behavior to work but didn't have time to implement it for now, so I opted to leave a skipped test in place as a reminder.

File: lib/magma/loader.rb [hunk 3]
```diff
@@ -25,16 +25,6 @@ def entry value
 
   # A generic loader class.
   class Loader
-    class << self
-      def description desc=nil
```
Comment (coleshaw) [inline lib/magma/loader.rb:29]: Removing this method seems to break the IPI project. I get:

```
magma  | 	15: from /home/developer/magma/lib/magma/server.rb:13:in `initialize'
magma  | 	14: from /home/developer/magma/lib/magma/server.rb:13:in `tap'
magma  | 	13: from /home/developer/magma/lib/magma/server.rb:14:in `block in initialize'
magma  | 	12: from /home/developer/magma/lib/magma.rb:52:in `load_models'
magma  | 	11: from /home/developer/magma/lib/magma.rb:52:in `each'
magma  | 	10: from /home/developer/magma/lib/magma.rb:53:in `block in load_models'
magma  | 	 9: from /home/developer/magma/lib/magma.rb:53:in `new'
magma  | 	 8: from /home/developer/magma/lib/magma/project.rb:29:in `initialize'
magma  | 	 7: from /home/developer/magma/lib/magma/project.rb:84:in `load_project'
magma  | 	 6: from /home/developer/magma/lib/magma/project.rb:92:in `load_project_files'
magma  | 	 5: from /home/developer/magma/lib/magma/project.rb:92:in `require'
magma  | 	 4: from /home/developer/magma/projects/ipi/requirements.rb:18:in `<top (required)>'
magma  | 	 3: from /home/developer/magma/projects/ipi/requirements.rb:18:in `require_relative'
magma  | 	 2: from /home/developer/magma/projects/ipi/loaders/flow_jo.rb:3:in `<top (required)>'
magma  | 	 1: from /home/developer/magma/projects/ipi/loaders/flow_jo.rb:4:in `<module:Ipi>'
magma  | /home/developer/magma/projects/ipi/loaders/flow_jo.rb:5:in `<class:FlowJoLoader>': undefined method `description' for Ipi::FlowJoLoader:Class (NoMethodError)
```

Are you using a branch of the IPI project, or no longer using it? Or is there something I need to tweak locally?

**Comment by coleshaw** [comment]
Outside of the issue with the `Loader` description method, looks good to me. Ran fine when I commented out the descriptions in each of the IPI loaders, but that should get addressed before deploying (either in Loader or in IPI), otherwise I worry the deployments will break.

**Pull request merged**